internal/exec/stages/disks/luks.go

```diff
@@ -531,3 +564,151 @@ func (d LuksDump) hasFlag(flag string) bool {
 	}
 	return false
 }
+
+// collect the cex card domains xx.xxxx
+func getAllCexOnlineAPQNs() (string, error) {
+	apDir := "/sys/bus/ap/devices/"
+	var ret string
+	// cards details are in form of link file points to card directory.
+	event, _ := os.ReadDir(apDir)
+	if len(event) == 0 {
+		return "", fmt.Errorf("adjunct processor not found")
+	}
+	// match the regexp for pattern xx.xxxx
+	for _, dir := range event {
+		match := cexRegx.MatchString(dir.Name())
+		// if a match is found, read the uevent from that APQN directory
+		// to verify whether it corresponds to a CCA controller. This is
+		// intended for potential future use with an EP11 controller.
+		if match {
+			filePath := filepath.Join(apDir, dir.Name(), "uevent")
+			ctrl, err := getCryptoController(filePath)
+			if err != nil {
+				return "", fmt.Errorf("failed to parse uevent file: %w", err)
+			}
+			for key := range ctrl {
+				if ctrl[key] == "cca" {
+					ret += dir.Name() + ","
```

Comment:
This will append a final comma at the end of `ret`. I think it'd be cleaner to make `ret` a `[]string{}` instead and then append to it here, and just at the end do `strings.Join()`.

Reply:
Done.

Earlier commit i used `strings.Join(ret, ",") , nil` with `[]string{}` ,  :D  you said it does not look good to use `strings.Join` that`s why  i re-wrote to string concatenate.


```
	for _, dir := range event {
		match := cexRegx.FindStringSubmatch(dir.Name())
		// if a match is found, read the uevent from that APQN directory
		// to verify whether it corresponds to a CCA controller. This is
		// intended for potential future use with an EP11 controller.
		if match != nil {
			filePath := apDir + strings.Join(match, "") + uEvent
			ctrl, err := getCryptoController(filePath)
			if err != nil {
				return "", fmt.Errorf("failed to parse uevent file: %w", err)
			}
			for key := range ctrl {
				if ctrl[key] == "cca" {
					ret = append(ret, match[0])
				}
			}
			if len(ret) == 0 {
				return "", fmt.Errorf("cannot find the cca ap controller")
			}
		}
	}
	return strings.Join(ret, ","), nil
}
```

Reply:
Yes sorry, you're right. I don't recall why I thought it'd be cleaner in that earlier conversation. Maybe I didn't think about the trailing comma issue.